core/base/inc/TROOT.h

```diff
@@ -297,6 +299,7 @@ friend TROOT *ROOT::Internal::GetROOT2();
    void              SaveContext();
    void              SetApplication(TApplication *app) { fApplication = app; }
    void              SetBatch(Bool_t batch = kTRUE) { fBatch = batch; }
+   void              SetRandomAccessCompression(Bool_t random = kTRUE) { fRandomAccessCompression = random; }
```

Comment:
Actually, yes ... unless we have a strong algorithm that when requested only turn it on in the case where it is beneficial ... to pass the information through TTree::Branch, the easiest might be to convolute the information into the splitlevel parameter.


For storage of the bit in memory, you should consider using SetBit/TestBit rather than a new data member.   (See the usage of kBranchAny or kCircular).